pkgs/by-name/op/opengamepadui/package.nix

```diff
@@ -0,0 +1,122 @@
+{
+  alsa-lib,
+  autoPatchelfHook,
+  cargo,
+  dbus,
+  fetchFromGitHub,
+  gamescope,
+  gitUpdater,
+  godot_4,
+  godot_4-export-templates,
+  hwdata,
+  lib,
+  libGL,
+  libpulseaudio,
+  mesa-demos,
+  pkg-config,
+  rustPlatform,
+  stdenv,
+  udev,
+  upower,
+  vulkan-loader,
+  xorg,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "opengamepadui";
+  version = "0.35.1";
+  withDebug = false;
+
+  buildType = "${if finalAttrs.withDebug then "debug" else "release"}";
+
+  src = fetchFromGitHub {
+    owner = "ShadowBlip";
+    repo = "OpenGamepadUI";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-741itXn39MxVMTNN9Azx2n5L1PRo5UsRsm6JSeY9nSI=";
+  };
+
+  cargoDeps = rustPlatform.fetchCargoVendor {
+    src = finalAttrs.src;
+    sourceRoot = "source/${finalAttrs.cargoRoot}";
+    hash = "sha256-sTzMewIfKHbmVhSPZgUIzFFz1ahK+PMoQ5oB4GEt8nY=";
+  };
+  cargoRoot = "extensions";
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    cargo
+    godot_4
+    godot_4-export-templates
+    pkg-config
+    rustPlatform.cargoSetupHook
+  ];
+
+  runtimeDependencies = [
+    alsa-lib
+    dbus
+    gamescope
+    hwdata
+    libGL
+    libpulseaudio
+    mesa-demos
+    udev
+    upower
+    vulkan-loader
+    xorg.libX11
+    xorg.libXcursor
+    xorg.libXext
+    xorg.libXi
+    xorg.libXrandr
+    xorg.libXres
+    xorg.libXtst
+  ];
+
+  dontStrip = finalAttrs.withDebug;
+
+  env = {
+    GODOT = lib.getExe godot_4;
+    EXPORT_TEMPLATE = "${godot_4-export-templates}";
+    BUILD_TYPE = "${finalAttrs.buildType}";
+  };
+
+  buildPhase = ''
+    runHook preBuild
+
+    GODOT_VERSION=$(echo "${godot_4.version}" | cut -d'-' -f1)
+    GODOT_RELEASE=$(echo "${godot_4.version}" | cut -d'-' -f2)
+
+    # Godot looks for export templates in HOME
+    export HOME=$(mktemp -d)
+    mkdir -p $HOME/.local/share/godot/export_templates
+    ln -s "${godot_4-export-templates}" "$HOME/.local/share/godot/export_templates/$GODOT_VERSION.$GODOT_RELEASE"
+
+    make build
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    make install PREFIX=$out
+
+    # The Godot binary looks in "../lib" for gdextensions
+    mkdir -p $out/share/lib
+    mv $out/share/opengamepadui/*.so $out/share/lib
```

Comment:
Could be postInstall, with `installFlags = [ "PREFIX=${placeholder "out"}" ];` (makeFlags works too)